src/as400_sha_fmt_plug.c

```diff
@@ -0,0 +1,419 @@
+// Check which of the calculated hashes matched
+
+static int as400_cmp_one(void *binary,int index)
+{
+	return !memcmp(as400_computed_binary_hash[index],as400_retrieved_binary_hash[index],BINARY_LENGTH);
+}
+
+
+// ==================================================================================================
+// No ASCII ciphertext, thus returns true
+
+static int as400_cmp_exact(char *source,int index)
+{
+	return 1;
+}
+
+
+// ==================================================================================================
+// Structure holding data for internal self tests
+
+static struct fmt_tests as400_tests[] =
+{
+	// Format of each line: {"hash like in internal representation", "valid password for hash", {"userid as in file","hash as in file"}}
+	{"ROB       4C106E52CA196986E1C52C7FCD02AF046B76C73C", "banaan", {"ROB","4C106E52CA196986E1C52C7FCD02AF046B76C73C"}},
+	{"BART      CED8050C275A5005D101051FF5BCCADF693E8AB7", "Kulach007", {"BART","CED8050C275A5005D101051FF5BCCADF693E8AB7"}},
+	{"SYSOPR    1BA6C7D54E9696ED33F4DF201E348CA8CA815F75", "T0Psecret!", {"SYSOPR","1BA6C7D54E9696ED33F4DF201E348CA8CA815F75"}},
+	{"SYSTEM    A1284B4F1BDD7ED598D4B5060D861D6D614620D3", "P@ssword01", {"SYSTEM","A1284B4F1BDD7ED598D4B5060D861D6D614620D3"}},
+	{"QSYSDBA   94C55BC7EDF1996AC62E8145CDBFA285CA79ED2E", "qsysdba", {"QSYSDBA","94C55BC7EDF1996AC62E8145CDBFA285CA79ED2E"}},
+	{"QSECOFR   CDF4063E283B51EDB7B9A8E6E542042000BD9AE9", "qsecofr!", {"QSECOFR","CDF4063E283B51EDB7B9A8E6E542042000BD9AE9"}},	
+	{"TEST1     44D43148CFE5CC3372AFD2610BEE3D226B2B50C5", "password1", {"test1","44D43148CFE5CC3372AFD2610BEE3D226B2B50C5"}},
+	{"TEST2     349B12D6588843A1632649A501ABC353EBF409E4", "secret", {"TEST2","349B12D6588843A1632649A501ABC353EBF409E4"}},
+	{"TEST3     A97F2F9ED9977A8A628F8727E2851415B06DC540", "test3", {"TeSt3","A97F2F9ED9977A8A628F8727E2851415B06DC540"}},
+	{NULL}
+};
+
+
+// ==================================================================================================
+// JtR as400 structure registration
+
+struct fmt_main fmt_as400 =
+{
+	{
+		FORMAT_LABEL,
+		FORMAT_NAME,
+		ALGORITHM_NAME,
+		BENCHMARK_COMMENT,
+		BENCHMARK_LENGTH,
+        0,
+		PLAINTEXT_LENGTH,
+		BINARY_SIZE,
+		BINARY_ALIGN,
+		SALT_SIZE,
+		SALT_ALIGN,
+		MIN_KEYS_PER_CRYPT,
+		MAX_KEYS_PER_CRYPT,
+		FMT_CASE | FMT_8_BIT | FMT_OMP,
+        { NULL },
+        { NULL },
+		as400_tests
+	}, {
+		as400_init,
+		fmt_default_done,
+		fmt_default_reset,
+		as400_prepare,
+		as400_valid,
+		fmt_default_split,
+		fmt_default_binary,
+		get_salt,
+		{ NULL },
+		fmt_default_source,
+		{
+			fmt_default_binary_hash
+		},
+		fmt_default_salt_hash,
+        NULL, 
+		set_salt,
+		as400_set_key,          
+		as400_get_key,          
+		fmt_default_clear_keys,
+		as400_crypt_all,        
+		{
+			fmt_default_get_hash
+		},
+		as400_cmp_all,          
+		as400_cmp_one,          
+		as400_cmp_exact
+	}
+};
+
+#endif /* plugin stanza */
```

Comment:
Does your editor support EditorConfig? If yes, it is easier to avoid the missing newline.